app/src/app/views/stix/collection/collection-import/collection-import-workflow/collection-import.component.ts

```diff
@@ -19,266 +22,544 @@ import { Tactic } from 'src/app/classes/stix/tactic';
-            switch (object.type) {
-                case "attack-pattern": //technique
-                    this.object_import_categories.technique[category].push(new Technique(raw))
-                break;
-                case "x-mitre-tactic": //tactic
-                    this.object_import_categories.tactic[category].push(new Tactic(raw))
-                break;
-                case "malware": //software
-                case "tool": 
-                    this.object_import_categories.software[category].push(new Software(object.type, raw))
-                break;
-                case "relationship": //relationship
-                    if (object.source_ref in idToSdo) raw.source_object = idToSdo[object.source_ref]
-                    if (object.target_ref in idToSdo) raw.target_object = idToSdo[object.target_ref]
-                    let rel = new Relationship(raw)
-                    this.object_import_categories.relationship[category].push(rel)
-                break;
-                case "course-of-action": //mitigation
-                    this.object_import_categories.mitigation[category].push(new Mitigation(raw))
-                break;
-                case "x-mitre-matrix": //matrix
-                    this.object_import_categories.matrix[category].push(new Matrix(raw))
-                break;
-                case "intrusion-set": //group
-                    this.object_import_categories.group[category].push(new Group(raw))
-                break;
-                case "x-mitre-data-source": // data source
-                    this.object_import_categories.data_source[category].push(new DataSource(raw))
-                break;
-                case "x-mitre-data-component": // data component
-                    this.object_import_categories.data_component[category].push(new DataComponent(raw))
-                break;
-                case "campaign": // campaign
-                    this.object_import_categories.campaign[category].push(new Campaign(raw))
-                break;
-            }
+        i.attack_id = (i.attack_id) ? i.attack_id : "";
+        i.description = (i.description) ? i.description : "";
+        i.type = (i.id) ? i.id.split('--')[0] : '';
+        i.x_mitre_version = (i.x_mitre_version) ? i.x_mitre_version.toString() : "1.0";
+        i.spec_version = '2.1';
+        i.x_mitre_is_subtechnique = Boolean(i.x_mitre_is_subtechnique);
+        i.created = i.created ? new Date(i.created).toISOString() : '';
+        i.modified = i.modified ? new Date(i.modified).toISOString() : '';
+        i.x_mitre_platforms = (i.x_mitre_platforms) ? i.x_mitre_platforms.split(', ') : [];
+        i.x_mitre_domains = (i.x_mitre_domains) ? i.x_mitre_domains.split(', ') : [];
+        i.x_mitre_data_sources = i.x_mitre_data_sources ? i.x_mitre_data_sources.split(',') : [];
+        i.external_references = [{
+            source_name: 'mitre-attack',
+            external_id: i.attack_id,
+          }];
```

Comment:
Similar comment here as below. In cases where the source object does not have an ATT&CK ID, the ATT&CK ID does not need an entry in the external references array.